frontend/src/components/teacher/student-management/AddOrEditStudentModal.tsx

```diff
@@ -96,28 +99,54 @@ const AddStudentModal = ({
     } else {
       setErrorMessage("");
 
-      await createStudent({
```

Comment:
courtesy of chatgpt - can make a bit more concise? up to you

```
const studentData = {
  firstName: watch("firstName"),
  lastName: watch("lastName"),
  studentNumber: watch("studentNumber"),
};

const operation = studentId ? updateStudent : createStudent;

await operation({
  variables: studentId
    ? { classId, studentId, student: studentData }
    : { student: studentData, classId },
})
  .then(() => {
    showToast({
      message: studentId ? "Student updated." : "New student created.",
      status: "success",
    });
  })
  .catch(() => {
    showToast({
      message: studentId
        ? "Failed to update student. Please try again."
        : "Failed to create a new student. Please try again.",
      status: "error",
    });
  });
```

Reply:
Agree this can definitely be more concise, but considering that I'm going to be refactoring the whole thing anyways when I do my modal refactor (coming up next after the auth stuff!), I think I'll defer it to then